iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/load/LoadTsFileToTableModelAnalyzer.java

```diff
@@ -0,0 +1,241 @@
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(LoadTsFileToTableModelAnalyzer.class);
+
+  private final Metadata metadata;
+
+  public LoadTsFileToTableModelAnalyzer(
+      LoadTsFileStatement loadTsFileStatement, Metadata metadata, MPPQueryContext context) {
+    super(loadTsFileStatement, context);
+    this.metadata = metadata;
+  }
+
+  public LoadTsFileToTableModelAnalyzer(
+      LoadTsFile loadTsFileTableStatement, Metadata metadata, MPPQueryContext context) {
+    super(loadTsFileTableStatement, context);
+    this.metadata = metadata;
+  }
+
+  @Override
+  public IAnalysis analyzeFileByFile(IAnalysis analysis) {
+    // check if the system is read only
+    if (CommonDescriptor.getInstance().getConfig().isReadOnly()) {
+      analysis.setFinishQueryAfterAnalyze(true);
+      analysis.setFailStatus(
+          RpcUtils.getStatus(TSStatusCode.SYSTEM_READ_ONLY, LoadReadOnlyException.MESSAGE));
+      return analysis;
+    }
+
+    try {
+      autoCreateDatabase(database);
+    } catch (VerifyMetadataException e) {
+      LOGGER.warn("Auto create database failed: {}", database, e);
+      analysis.setFinishQueryAfterAnalyze(true);
+      analysis.setFailStatus(RpcUtils.getStatus(TSStatusCode.LOAD_FILE_ERROR, e.getMessage()));
+      return analysis;
+    }
+
+    if (!doAnalyzeFileByFile(analysis)) {
+      // return false means the analysis is failed because of exception
+      return analysis;
+    }
+
+    LOGGER.info("Load - Analysis Stage: all tsfiles have been analyzed.");
+
+    // data partition will be queried in the scheduler
+    setRealStatement(analysis);
+    return analysis;
+  }
+
+  @Override
+  protected void analyzeSingleTsFile(final File tsFile) throws IOException, AuthException {
+    try (final TsFileSequenceReader reader = new TsFileSequenceReader(tsFile.getAbsolutePath())) {
+      // can be reused when constructing tsfile resource
+      final TsFileSequenceReaderTimeseriesMetadataIterator timeseriesMetadataIterator =
+          new TsFileSequenceReaderTimeseriesMetadataIterator(reader, true, 1);
+
+      // check if the tsfile is empty
+      if (!timeseriesMetadataIterator.hasNext()) {
+        throw new LoadEmptyFileException(tsFile.getAbsolutePath());
+      }
+
+      // check whether the tsfile is table-model or not
+      if (Objects.isNull(reader.readFileMetadata().getTableSchemaMap())
+          || reader.readFileMetadata().getTableSchemaMap().size() == 0) {
+        throw new SemanticException("Attempted to load a tree-model TsFile into table-model.");
+      }
+
+      // construct tsfile resource
+      final TsFileResource tsFileResource = constructTsFileResource(reader, tsFile);
+
+      for (Map.Entry<String, TableSchema> name2Schema :
+          reader.readFileMetadata().getTableSchemaMap().entrySet()) {
+        org.apache.iotdb.db.queryengine.plan.relational.metadata.TableSchema realSchema =
+            metadata
+                .validateTableHeaderSchema(
```

Comment:
TODO: handle cases when there exists a table whose metadata is not consistent with the loading file